dev-tools/check_duplicate_test_data.php

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env php
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+require_once __DIR__.'/../vendor/autoload.php';
+
+$testClassNames = array_filter(
+    array_keys(require __DIR__.'/../vendor/composer/autoload_classmap.php'),
+    static fn (string $className): bool => str_starts_with($className, 'PhpCsFixer\Tests\\')
+);
+
+if ([] === $testClassNames) {
+    echo 'Run: composer dump-autoload --optimize --working-dir=', realpath(__DIR__.'/..'), PHP_EOL;
+
+    exit(1);
+}
+
+$duplicatesFound = false;
+
+foreach ($testClassNames as $testClassName) {
+    if (!str_ends_with($testClassName, 'Test')) {
+        continue;
+    }
+
+    foreach ((new ReflectionClass($testClassName))->getMethods() as $method) {
+        if (!$method->isPublic()) {
+            continue;
+        }
+        if ($method->getDeclaringClass()->getName() !== $testClassName) {
+            continue;
+        }
+        if (!str_starts_with($method->getName(), 'provide')) {
+            continue;
+        }
+
+        $exceptionsBecauseOfSerialization = [ // should only shrink
+            'PhpCsFixer\Tests\AutoReview\CommandTest::provideCommandHasNameConstCases',
```

Comment:
@kubawerlos' arguments are fine for me, many imports are not the problem for me, but since this method won't be navigable anyway, it's probably better to keep explicit FQMN.

Reply:
`PhpCsFixer\Tests\AutoReview\CommandTest::provideCommandHasNameConstCases(...)` can be used as well if no old PHP support is required.

Reply:
happy with any. I would rather focus to reduce the size of this array and get rid of it one day